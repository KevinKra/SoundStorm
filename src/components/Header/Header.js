import React, { Component } from "react";
import "./Header.scss";

export default class Header extends Component {
  render() {
    return (
      <header className="Header">
        <h2>TUNIC</h2>
        <nav>ROCK HIPHOP POP</nav>
      </header>
    );
  }
}
