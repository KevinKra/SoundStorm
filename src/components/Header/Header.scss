@import "../../_styles/variables.scss";

.Header {
  display: flex;
  flex-direction: column;
  // border: 1px solid red;
  width: 100%;
  height: 8vh;
  h2 {
    margin: 0;
    height: 80%;
  }
  nav {
    // border: 1px solid red;
    display: flex;
    justify-content: space-evenly;
  }
  button {
    background: none;
    border: none;
    color: $text-primary;
    font-size: 12px;
    &:hover {
      cursor: pointer;
      color: white;
    }
    &:focus {
      outline: none;
    }
  }
}
