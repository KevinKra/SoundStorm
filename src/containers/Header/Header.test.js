import React from "react";
import Header from "./Header";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";

describe("<Header />", () => {
  let wrapper;
  let store;
  const mockStore = configureStore();
  const initialState = [];
  const mockUpdatePrimaryTrack = jest.fn();
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Header store={store} />);
  });
  it("should match snapshot", () => {
    console.log(wrapper.debug());
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  xit("should call the updatePrimaryTrack function when button is clicked", () => {
    wrapper.find("button").simulate("click");
    expect(mockUpdatePrimaryTrack).toHaveBeenCalled();
  });
});
