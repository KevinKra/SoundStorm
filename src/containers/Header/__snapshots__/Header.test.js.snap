// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<Header
  loadPrimaryTracks={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
